src/components/expenses/expenses.service.js

```diff
@@ -0,0 +1,50 @@
+let expenses = [];
+
+export function getAll(userId, category, from, to) {
+  expenses = expenses.filter((expense) => expense.userId === userId);
+  expenses = expenses.filter((expense) => expense.category === category);
+  expenses = expenses.filter((expense) => expense.spentAt >= from);
+  expenses.expenses.filter((expense) => expense.spentAt <= to);
```

Comment:
you can use one filter here, because now it looks strange and it has bad performance